pandas/tests/base/test_misc.py

```diff
@@ -201,3 +201,22 @@ def test_get_indexer_non_unique_dtype_mismatch():
     indexes, missing = pd.Index(["A", "B"]).get_indexer_non_unique(pd.Index([0]))
     tm.assert_numpy_array_equal(np.array([-1], dtype=np.intp), indexes)
     tm.assert_numpy_array_equal(np.array([0], dtype=np.int64), missing)
+
+
```

Comment:
I might put it in `pandas/tests/indexes/test_base.py`, next to
 https://github.com/pandas-dev/pandas/blob/bdcc5bffaadb7488474b65554c4e8e96a00aa4af/pandas/tests/indexes/test_base.py#L2590-L2616